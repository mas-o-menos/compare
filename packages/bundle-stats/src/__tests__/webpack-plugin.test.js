const path = require('path');
const webpack = require('webpack');
const MemoryFS = require('memory-fs');
const { BundleStatsWebpackPlugin } = require('../webpack-plugin');
const { merge } = require('lodash');

jest.setTimeout(10 * 1000);

const BASE_CONFIG = {
  mode: 'production',
  context: path.join(__dirname, '../../test/package/app'),
};

describe('webpack plugin', () => {
  test('default config', (done) => {
    expect.assertions(3);

    const compiler = webpack(merge({}, BASE_CONFIG, {
      plugins: [
        new BundleStatsWebpackPlugin(),
      ],
    }));
    compiler.outputFileSystem = new MemoryFS()

    compiler.run((error, stats) => {
      expect(error).toEqual(null);
      expect(stats.hasErrors()).toBe(false);
      expect(stats.toJson({ source: false, assets: true }).assets).toMatchSnapshot();
      done();
    });
  });

  test('baseline', (done) => {
    expect.assertions(3);

    const compiler = webpack(merge({}, BASE_CONFIG, {
      plugins: [
        new BundleStatsWebpackPlugin({ baseline: true }),
      ],
    }));
    compiler.outputFileSystem = new MemoryFS()

    compiler.run((error, stats) => {
      expect(error).toEqual(null);
      expect(stats.hasErrors()).toBe(false);
      expect(stats.toJson({ source: false, assets: true }).assets).toMatchSnapshot();
      done();
    });
  });
});

