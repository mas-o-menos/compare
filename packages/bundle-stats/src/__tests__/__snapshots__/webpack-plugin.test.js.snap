// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack plugin baseline 1`] = `
Array [
  Object {
    "chunkNames": Array [],
    "chunks": Array [],
    "emitted": true,
    "isOverSizeLimit": undefined,
    "name": "../node_modules/.cache/bundle-stats/baseline.json",
    "size": 0,
  },
  Object {
    "chunkNames": Array [],
    "chunks": Array [],
    "emitted": true,
    "isOverSizeLimit": undefined,
    "name": "bundle-stats.html",
    "size": 0,
  },
  Object {
    "chunkNames": Array [
      "main",
    ],
    "chunks": Array [
      0,
    ],
    "emitted": true,
    "isOverSizeLimit": undefined,
    "name": "main.js",
    "size": 956,
  },
]
`;

exports[`webpack plugin default config 1`] = `
Array [
  Object {
    "chunkNames": Array [],
    "chunks": Array [],
    "emitted": true,
    "isOverSizeLimit": undefined,
    "name": "bundle-stats.html",
    "size": 0,
  },
  Object {
    "chunkNames": Array [
      "main",
    ],
    "chunks": Array [
      0,
    ],
    "emitted": true,
    "isOverSizeLimit": undefined,
    "name": "main.js",
    "size": 956,
  },
]
`;
